### ISSUE TITLE
git can't pull from http repositories when executed on buildFHSUserEnv

### ISSUE THREAD

**Issue by FranciscoKurpiel** [NixOS/nixpkgs#210223]
### Describe the bug
When building a bazel project that includes on WORKSPACE a `git_repository` with an https url:

Git fails to clone https repositories when running with buildFHSUserEnv, even when `cacert` is included. This is necessary to build go projects with bazel, if WORKSPACE have a reference to a repository using https.

### Steps To Reproduce
shell.nix:
```
# vim: set expandtab sw=2 ts=2:
{ pkgs ? import <nixpkgs> {} }:

(pkgs.buildFHSUserEnv {
  name = "bazel-build";
  targetPkgs = pkgs: with pkgs; [
    cacert
    git
  ];
}).env
```

```bash
nix-shell --pure
git clone https://github.com/bazelbuild/rules_go.git
```

### Expected behavior
Repository is cloned.

### Screenshots
![image](https://user-images.githubusercontent.com/11741077/211855266-330f2ce9-2889-4864-831d-4fc1a6558bcc.png)

### Additional context
This is a minimum reproducible procedure. The actual need is to work with bazel on a project that includes go, and WORKSPACE file has references to git repository using https. Building has to be done with `buildFHSUserEnv` and `--pure`, and the git command line will be used to try to clone the repository.

The WORKSPACE that originally caused the issue includes:
```bazel
git_repository(
    name = "io_bazel_rules_go",
    remote = "https://github.com/bazelbuild/rules_go.git",
    commit = "8d309d5af15814b4096d80b60f80fa86128c43f2"
)
```

When building something that uses it, the reported git error happens.

### Notify maintainers
@mboes @libjared

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.3, NixOS, 23.05 (Stoat), 23.05pre441271.c07552f6f7d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - channels(root): `"nixos"`
 - channels(xyko): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by libjared** [comment]
**tl;dr**: This is a known side-effect (IDK if I'd call it a "bug") of pure shells, the usual workaround doesn't apply with FHS.

I understand this issue has, at its core, nothing to do with Golang or Bazel.

I can reproduce the "unregistered scheme" error. If I also include `curl` into the FHSUserEnv, and try `curl https://example.com`, I get

> curl: (77) error setting certificate file: /no-cert-file.crt

So this issue _also_ has nothing to with Git; it's caused entirely by the lack of access to SSL certs in pure nix-shells. This is a known pain point. This is the same issue as #66716, except with one major distinction: The workaround, adding `cacert` to the pure shell, does not work with `buildFHSUserEnv`.

Why is that? Well in a normal pure nix-shell, adding `cacert` sets 3 env vars `SYSTEM_CERTIFICATE_PATH` `NIX_SSL_CERT_FILE` `SSL_CERT_FILE` thanks to its [setup-hook.sh](https://github.com/NixOS/nixpkgs/blob/a83ed85c14fcf242653df6f4b0974b7e1c73c6c6/pkgs/data/misc/cacert/setup-hook.sh). But this does not carry into the FHSUserEnv, these variables are unchanged within.

So, the workaround now involves passing these variables from the pure nix-shell into the FHSUserEnv.

@FranciscoKurpiel Can you try specifying a custom `runScript` like how [quartus-prime](https://github.com/NixOS/nixpkgs/blob/6fecd5a257327db6116b28acdfb45561e65e3dfa/pkgs/applications/editors/quartus-prime/default.nix#L89-L91) or [dropbox](https://github.com/NixOS/nixpkgs/blob/6fecd5a257327db6116b28acdfb45561e65e3dfa/pkgs/applications/networking/dropbox/default.nix#L48-L49) does it?

**Comment by FranciscoKurpiel** [comment]
I was aware of #66716, but was caught by the change of behavior of including the `pkgs.cacert` on a _fhs_ environment. I find this very surprising, and I'm finding very hard to see it as anything but a bug. It looks to me like this feature has quite a few sharp edges. But with your pointers I was able work around the toy example I provided:

```nix
{ pkgs ? import <nixpkgs> {} }:

(pkgs.buildFHSUserEnv {
  name = "bazel-build";
  targetPkgs = pkgs: with pkgs; [
    cacert
    git
  ];
  profile = ''
    export SSL_CERT_FILE=${pkgs.cacert}/etc/ssl/certs/ca-bundle.crt
    export SYSTEM_CERTIFICATE_PATH="$SSL_CERT_FILE"
    export GIT_SSL_CAINFO="$SSL_CERT_FILE"
  '';
}).env
```

With this I can start a pure FHS shell and use curl/git with TLS, and it looks hermetic to me. At a first glance it looks like the issue I was having with Bazel is also fixed, so I'm closing the issue. Thank-you very much for your time.
